### ISSUE TITLE
Don't use orderless with company-capf as a default

### ISSUE THREAD

**Issue by niontrix** [syl20bnr/spacemacs#16302]
Hi guys!
I recently switched from using helm to the compleseus-layer which uses orderless for completion styles. So far it works great.
But a few days ago I noticed that Company still seems to have an unresolved issue that causes the auto-completion to become unbearably slow when using ordereless as a completion style.
Currently the compleseus-layer configures orderless to not be used as a completion style when lsp-completion-mode is active, because as far as I know lsp provides its own completion styles, so its not needed.
The problem I have is that when I try to use auto-completion in an emacs-lisp buffer the auto-completion gets unuseably slow, which I could trace back to the use of orderless, using the profiler.
Like in https://github.com/oantolin/orderless/issues/48#issuecomment-856750410 I think it maybe best not to use orderless as a completion style with company-capf by default, because it seems to be a general issue with company when using orderless, so I expect it to cause the same problems in other modes as well.

https://github.com/syl20bnr/spacemacs/blob/4a227fc94651136a8de54bcafa7d22abe1fa0295/layers/%2Bcompletion/compleseus/packages.el#L300-L310

Proposed:
```elisp
 (defun compleseus/init-orderless () 
   (use-package orderless 
     :init 
     ;; company-capf is messed up with orderless 
     ;; https://github.com/oantolin/orderless/issues/48#issuecomment-856750410 
     (define-advice company-capf (:around (orig-fun &rest args) set-completion-styles) 
       (let ((completion-styles '(basic partial-completion orderless))) 
         (apply orig-fun args)))
```

**Comment by smile13241324** [comment]
Sounds reasonable I'll put it in my queue

**Comment by fnussbaum** [comment]
> The problem I have is that when I try to use auto-completion in an emacs-lisp buffer the auto-completion gets unuseably slow

I encountered the same problem, but it was not due to orderless, as it persisted with only the basic completion style enabled on my system. I looked into it, and it was mainly due to undo-tree polluting `obarray` with hundreds of thousands garbage symbols `undo-tree-id...`. The following function counted more than a million for me:
```elisp
(defun fn/count-undo-tree-ids ()
  (interactive)
  (let ((count 0))
    (cl-do-all-symbols (sym)
      (when (string-prefix-p "undo-tree-id" (symbol-name sym))
        ;; (unintern sym) ; this fixes the slowness for me
        (cl-incf count)))
    (message "Counted %s undo-tree-id symbols in obarray." count)))
```
Those symbols were initially created to deal with markers and should have stayed uninterned, but they get interned implicitly during deserialisation. In particular they do not serve any purpose after deserialisation, as `undo-tree-deserialize` initialises the object-pool to an empty hash table, and markers do not survive having their buffer killed anyway. 

The completion using company still felt a little bit slow, hence I switched to corfu for `emacs-lisp-mode` which unlike company does not seem to slow down typing at all, and one can easily use orderless with a SPC separator.

> Sounds reasonable I'll put it in my queue

I think the existing advice is fine, at least on my system disabling orderless does not seem to have any benefits.
